### ISSUE TITLE
Board freezes after making many touches at once

### ISSUE THREAD

**Issue by Porizm** [jsxgraph/jsxgraph#374]
Hello,

Please see this gif for demonstration. Sometimes it doesn't freeze, but board becomes zoom in and out using only one finger instead of pan.

![IMG_5mlcxt](https://user-images.githubusercontent.com/2933548/131193898-6eab288c-a755-47c4-a256-49d12ad15650.gif)

**Comment by alfredwassermann** [comment]
Does the freeze happen after you touched the screen with your palm? I could not force a freeze on my devices, so far.
Can you please open https://jsfiddle.net/upksjr72/ with your smartphone and give feedback which kind of events are used by your browser (the text will show "pointer events, touch events or mouse events")? 
Further, can you give the board attributes which are used?

**Comment by Porizm** [comment]
Hello,

In the jsfiddle, it says "pointer events".

And these are the attributes I used in that gif demo:

		this.board = JXG.JSXGraph.initBoard('container', {
			boundingbox: boundingBox,
			axis: true,
			showcopyright: false,
			showNavigation: false,
			keepaspectratio: true,
			grid: false,
			pan: {
				enabled: true,
				needTwoFingers: false,
				needShift: false,
			},
			selection: {
				enabled: false
			},
			resize: { enabled: false },
			zoom: {
				factorX: 1.25,
				factorY: 1.25,
				pinchHorizontal: false,
				pinchVertical: false,
				wheel: false,
				needShift: false,
			},
			keyboard: {
				enabled: false
			},
			resize: {
				enabled: false
			}
		});

**Comment by Porizm** [comment]
Catching this bug on android is easier than iOS

**Comment by alfredwassermann** [comment]
Meanwhile, I can produc a similar bug. It will be repaired next week. Thank you very much for pointing it out!

**Comment by alfredwassermann** [comment]
The freeze that I could produce is now fixed in the sources. Could you please test tonight's nightly build (after 10pm CEST) if it also fixes your problems?
Thanks a lot, Alfred

**Comment by Porizm** [comment]
I just downloaded "2021-09-13.2000" which I think it's the last version.

Sadly, the issue is still happening

**Comment by alfredwassermann** [comment]
Unfortunately, I still can not reproduce this bug as you did in your gif movie. Could you describe what you are doing exactly to freeze the board?

**Comment by Porizm** [comment]
![palm](https://user-images.githubusercontent.com/2933548/133301389-e6503aed-0184-4dd3-8396-8a9affaa4bc8.png)

The issue occurs only when those 2 areas touches the screen. They must touch screen at same time then swipe right, I do it all the time with my phone to quickly clean the screen using right hand.

I think the reason is because doing this is passing 4 touches at the same time to event, and their values are confusing the library.

**Comment by alfredwassermann** [comment]
Next try. With my old fingers I can still not reproduce this bug. However today's nightly built contains a fix which resets JSXGraph's bookkeeping of the fingers whenever a finger fires the down event and it is the first finger on the device.

**Comment by Porizm** [comment]
Thank you very much. I will try the build tonight.
If it's too hard to reproduce the bug, then it's at least a good thing.

**Comment by Porizm** [comment]
I just tried the last version. issue is still.

I don't think it's critical issue because it's hard to find.

Thank you for your effort.

**Comment by alfredwassermann** [comment]
Thank you so much for work! If you still have energy to find this bug then I have one more suggestion: We just released v1.3.1. You can download the file `jsxgraphsrc.js` from the `distrib` folder and use this file instead of `jsxgraphcore.js`. It is the uncompressed version. Then connect your smartphone to a chrome browser on a laptop / desktop computer (see e.g. https://developer.chrome.com/docs/devtools/remote-debugging/ for help) and have a look what kind of errors are thrown when the board freezes.

Best wishes, Alfred

**Comment by Porizm** [comment]
I did those instructions and nothing was logged in console. But I noticed something that might help.

in function pointerDownListener I found this:

            if (this.attr.drag.enabled && object) {
                elements = [ object ];
                this.mode = this.BOARD_MODE_DRAG;
            } else {
                elements = this.initMoveObject(pos[0], pos[1], evt, type);
            }
I logged "this.mode" after that condition.

When touching with one finger, this.mode is 0
2 fingers > this.mode is 2
3 or more > 17

but when that issue occurs and the board freezes, this.mode is always 17. maybe this.mode needs to be reset or something.

Thank you

**Comment by alfredwassermann** [comment]
OK, this is very helpful: 17 is zoom mode. I submitted a few changes which might cure the disease in today's nightly build.

When you watched the board freeze in the console, did you notice any pointer events after the freeze? If not, is the whole window frozen and sucks up CPU? 
What happens if you do not touch the phone for a few minutes? Does it ask if you want to stop the process (i.e. an infinite loop)?

Thanks a lot for devoting your time!

**Comment by Porizm** [comment]
The tab doesn't freeze at all. I think the reason why this is not happening in your device because maybe your device is fast, i have Samsung A20 which is considered slow, while it's hard to find the bug on iphone 7 because it's faster.

I logged all properties for board before and after the bug, hope this helps.

https://www.equations.me/jsx/beforeBug.json

https://www.equations.me/jsx/afterBug.json

sorry for putting them in my website, because they are big.

Thank you for the support.

**Comment by alfredwassermann** [comment]
Thank you very much, these logs are helpful! Unfortunately, the nightly build were broken because of a napm conflict. Cout you please test the problem with the second nightly build from Spetember 18th again? Sorry...

**Comment by Porizm** [comment]
Hello,

I tried the last build now that is 1.3.1 now. Zoom stopped working on device.
And the issue is still there, but something is different now and I think you are too close to fix it.
When the board is frozen, I did the palm touch again and it went back to normal, but zoom doesn't work.

One more thing, 

I used the old version again and went to function called pointerUpListener,

at the end of that function just before `return true;` added this `this._board_touches = []`

Hope this doesn't ruin something else, but it's half solution of the problem. When the board is frozen, i touch again and the board is no longer frozen.

Thank you
